projectgenerator/libqgsprojectgen/dataobjects/project.py

```diff
@@ -97,7 +97,7 @@ def create(self, path: str, qgis_project: QgsProject):
         qgis_relations = list(
             qgis_project.relationManager().relations().values())
         dict_domains = {
-            layer.real_id: layer.is_domain for layer in self.layers}
+            layer.layer.id(): layer.is_domain for layer in self.layers}
```

Comment:
I had to avoid `real_id` in this line because I was getting this nasty error:

```
SystemError: error return without exception set
Traceback (most recent call last):
  File "/home/germap/.local/share/QGIS/QGIS3/profiles/default/python/plugins/projectgenerator/gui/generate_project.py", line 290, in accepted
    project.create(None, qgis_project)
  File "/home/germap/.local/share/QGIS/QGIS3/profiles/default/python/plugins/projectgenerator/libqgsprojectgen/dataobjects/project.py", line 100, in create
    layer.real_id: layer.is_domain for layer in self.layers}
  File "/home/germap/.local/share/QGIS/QGIS3/profiles/default/python/plugins/projectgenerator/libqgsprojectgen/dataobjects/project.py", line 100, in 
    layer.real_id: layer.is_domain for layer in self.layers}
  File "/home/germap/.local/share/QGIS/QGIS3/profiles/default/python/plugins/projectgenerator/libqgsprojectgen/dataobjects/layers.py", line 111, in real_id
    if self.__layer:
SystemError: error return without exception set
```

Anyways, `layer.layer` at this point is already a `QgsVectorLayer`, so we don't need to do the check that the property `real_id` does internally.

Reply:
This looks very strange and it will be hard to know in the future, why this workaround was introduced.

Can we embed the call to `layer.real_id` in a `try: ...  except SystemError:` block and only revert to the other behavior when this actually happens? I think this way it will be more transparent what's going on and we can revert easier in the future.

Reply:
As I told you, at this point `layer.layer` is already a `QgsVectorLayer`, so we can access its methods without trouble.

This simple dict comprehension will become a bit ugly inside a try catch... but, ok.

BTW, no clue about the `SystemError`. Also, an Internet search didn't lead me to anything useful.

Reply:
What will happen if this is in the future - for whatever reason - called in a different context? Will we at least get a reasonable error in such a scenario?

Reply:
Since we above call:

```
for layer in self.layers:
    qgis_layer = layer.create()
```

at the point we use `layer.layer.id()`, `layer.layer` exists. We shouldn't get errors there. 

On the contrary, with the current situation a very strange error is happening. 

The change we're making in this PR avoids such strange error. I'll comment in that line why we changed from `layer.real_id` to `layer.layer.id()`.

Reply:
Alright, calling `layer.create()` is accepted as a reason to leave this as is.
(at the same time *strange error* is not accepted as a reason because *strange errors* tend to come back at one later on and bite one in the back.) ;)

Reply:
Yeah, since the first comment it was stated that `layer.layer` was already a `QgsVectorLayer`, so the change was safe anyways.